challenge.rb

```diff
@@ -1,26 +1,26 @@
 
 def capitalize_each_string(input)
-  #implement your solution here
+  input.map { | animal | animal.capitalize }
 end
 
 def fetch_the_dog(input)
-  #implement your solution here
+  input.find_all { |animal| animal.downcase == "dog" }
 end
 
 def no_dogs_allowed(input)
-  #implement your solution here
+  input.reject { |animal| animal.downcase == "dog" }
 end
 
 def count_the_animals(input)
-  #implement your solution here
+  input.count
 end
 
 def fetch_the_first_two(input)
-  #implement your solution here
+  input.first(2)
 end
 
 def fetch_CD_animals(input)
-  #implement your solution here
+  input.select { | animal | animal.downcase.start_with?("c", "d") }
```

Comment:
ruby has a cool enumerable called `.grep` which is handy here for particular use cases like this one.

https://ruby-doc.org/core-2.5.0/Enumerable.html#method-i-grep

Reply:
@jaybobo Something more like this `input.grep(/^[c,d]/i)` ?